### PR TITLE
Set up agent tests

### PR DESCRIPTION
**Pull request by kezhenxu94**

### CODE REVIEW

File: modules/nodejs-agent/test/plugins/http/expected.data.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+segmentItems:
+  - serviceName: test-http
+    segmentSize: 1
+    segments:
+      - segmentId: 1.248349.185910205007481
+        spans:
```
Comment (wu-sheng) [inline modules/nodejs-agent/test/plugins/http/expected.data.yml:23]: I think we missed the ref check here. And e2e should include the client/server spans

Comment (kezhenxu94) [reply modules/nodejs-agent/test/plugins/http/expected.data.yml:23]: > I think we missed the ref check here. And e2e should include the client/server spans

It’s plugin tests, not e2e

Comment (wu-sheng) [reply modules/nodejs-agent/test/plugins/http/expected.data.yml:23]: Yes, this is not e2e test, I mean, the client to server rpc. We should check the header propagation and ref data.

**Review by wu-sheng — APPROVED** [review]

**Comment by wu-sheng** [comment]
One question, how to add a new plugin test case?

**Comment by kezhenxu94** [comment]
> One question, how to add a new plugin test case?

Docs are added, nearly the same as the Python agent

**Pull request merged**